plaidml/bridge/openvino/ops/deformable_convolution.cpp

```diff
@@ -0,0 +1,290 @@
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+using namespace edsl;
+
+namespace {
+
+// Compute pad_before and the size of output.
+// This function is based on (the functon compute_padding_and_output_size() in plaidml/op/lib/ops.cc.).
+// The type of the return variable and parameters are changed from TensorDim to size_t, because TensorDim can't be
+// converted to int.
+std::pair<size_t, size_t> compute_padding_before_and_output_size(size_t input_size, size_t off_size, size_t filter_size,
+                                                                 size_t stride, plaidml::op::AutoPadMode autopad_mode,
+                                                                 size_t pad_before, size_t pad_end, size_t dilation) {
+  size_t I_eff = input_size;
+  size_t F_eff = (dilation * (filter_size - 1)) + 1;
+  if (autopad_mode == plaidml::op::AutoPadMode::EXPLICIT) {
+    size_t output_size = ((I_eff + pad_before + pad_end - F_eff + stride) / stride);
+    std::pair<size_t, size_t> result(pad_before, output_size);
+    return result;
+  }
+  if (autopad_mode == plaidml::op::AutoPadMode::VALID) {
+    size_t output_size = ((I_eff - F_eff + stride) / stride);
+    std::pair<size_t, size_t> result(0, output_size);
+    return result;
+  }
+  if (autopad_mode == plaidml::op::AutoPadMode::SAME_LOWER || autopad_mode == plaidml::op::AutoPadMode::SAME_UPPER) {
+    size_t lower_term = (autopad_mode == plaidml::op::AutoPadMode::SAME_LOWER) ? 1 : 0;
+    size_t max = 0;
+    if (((off_size - 1) * stride + F_eff - input_size) > 0) {
+      max = (off_size - 1) * stride + F_eff - input_size;
+    }
+    pad_before = (max + lower_term) / 2;
+    size_t output_size = ((I_eff + stride - 1) / stride);
+    std::pair<size_t, size_t> result(pad_before, output_size);
+    return result;
+  }
+  THROW_IE_EXCEPTION << "Unexpected autopadding mode.";
+}
+
+// Cast long int to int in vector.
+std::vector<int> cast_vector(std::vector<int64_t> vec) {
+  std::vector<int> cast_vec(vec.size());
+  for (auto i = 0; i < vec.size(); ++i) {
+    cast_vec[i] = static_cast<int>(vec[i]);
+  }
+  return cast_vec;
+}
+
+// Compute DeformableConvolution.
+edsl::Tensor compute_deformable_convolution(edsl::Tensor I, edsl::Tensor OFF, edsl::Tensor F, std::vector<int> I_shape,
+                                            std::vector<int> OFF_shape, std::vector<int> F_shape, int G, int DG,
+                                            size_t rank, std::vector<size_t> strides, std::vector<size_t> dilations,
+                                            std::vector<size_t> pad_befores) {
+  // Define dim of each tensor.
+  int N = I_shape[0];
+  int CI = I_shape[1];
+  int CO = F_shape[1];
+  int OFF_C = OFF_shape[1];
+  // Throw exception
+  if (CI % G != 0 || CO % G != 0 || CI % DG != 0 || OFF_C % DG != 0) {
+    THROW_IE_EXCEPTION << "Incorrect shape for DeformableConvolution.";
+  }
+  // Define new dimention.
+  std::vector<int> NEW_DIM;
```

Comment:
Please rename NEW_DIM to something more descriptive.

Reply:
what about Deformed_Dim or DEFORMED_DIM?I think Deformed_Dim is better.

Reply:
It is good.

Reply:
> what about Deformed_Dim or DEFORMED_DIM?I think Deformed_Dim is better.

 I think deformed_dims follows the naming convention.